homeassistant/components/media_player/emby.py

```diff
@@ -4,252 +4,227 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/media_player.emby/
 """
+import asyncio
 import logging
 
-from datetime import timedelta
-
 import voluptuous as vol
 
-import homeassistant.helpers.config_validation as cv
 from homeassistant.components.media_player import (
-    MEDIA_TYPE_TVSHOW, MEDIA_TYPE_VIDEO, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE,
-    SUPPORT_SEEK, SUPPORT_STOP, SUPPORT_PREVIOUS_TRACK, MediaPlayerDevice,
-    SUPPORT_PLAY, PLATFORM_SCHEMA)
+    MEDIA_TYPE_TVSHOW, MEDIA_TYPE_VIDEO, MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE, SUPPORT_SEEK, SUPPORT_STOP, SUPPORT_PREVIOUS_TRACK,
+    MediaPlayerDevice, SUPPORT_PLAY, PLATFORM_SCHEMA)
 from homeassistant.const import (
-    CONF_HOST, CONF_API_KEY, CONF_PORT, CONF_SSL, DEVICE_DEFAULT_NAME,
-    STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING, STATE_UNKNOWN)
-from homeassistant.helpers.event import (track_utc_time_change)
-from homeassistant.util import Throttle
+    STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING,
+    CONF_HOST, CONF_PORT, CONF_SSL, CONF_API_KEY, DEVICE_DEFAULT_NAME,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
 import homeassistant.util.dt as dt_util
 
-REQUIREMENTS = ['pyemby==0.2']
+REQUIREMENTS = ['pyemby==1.0']
+
+_LOGGER = logging.getLogger(__name__)
 
-MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
-MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+CONF_AUTO_HIDE = 'auto_hide'
 
 MEDIA_TYPE_TRAILER = 'trailer'
+MEDIA_TYPE_GENERIC_VIDEO = 'video'
 
+DEFAULT_HOST = 'localhost'
 DEFAULT_PORT = 8096
+DEFAULT_SSL = False
```

Comment:
Hoping we can have some more discussion on this.  Main issue is the default SSL configuration for Emby results in a connection without a valid certificate.  I can work around this on the library side of things, but the media image url's passed to hass will not be rendered with an invalid certificate (Tons of log errors are generated as a result).

This means that the average user setting up Emby and HASS for the first time will end up with a broken configuration using the defaults.  I totally understand the desire to opt-out of security, but in this case the out-of-the-box Emby is really not setup to properly use SSL without additional configuration that most users won't bother doing.

I would switch to the CONF_PROXY_SSL parameter, but I'm not sure how that's valid since there is no proxy involved here.